keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ResourceEvent.kt

```diff
@@ -238,22 +239,27 @@ data class ResourceActuationPaused(
  * Actuation on the managed resource has been vetoed.
  *
  * @property reason The reason why actuation was vetoed.
+ * @property veto The veto that vetoed this resource
  */
 data class ResourceActuationVetoed(
   override val kind: ResourceKind,
   override val id: String,
   override val application: String,
   val reason: String?,
```

Comment:
Updated to `veto`, not using that field anymore, but I think it's really useful so I kept it.
Added `suggestedStatus`, which is set by the veto, so that it can tell you what it thinks the status should be (and the `ResourceStatusService` has the final say or can override it. I think this is a lot less brittle!